### ISSUE TITLE
[receiver/datadog] `span.Type` and `span.Meta["span.kind"]` is different.

### ISSUE THREAD

**Issue by jpkrohling** [open-telemetry/opentelemetry-collector-contrib#18190]
`span.Type` and `span.Meta["span.kind"]` is different.

 `span.Type` comes from https://github.com/DataDog/dd-trace-go/blob/main/ddtrace/ext/app_types.go while `span.Meta["span.kind"]` comes from https://github.com/DataDog/dd-trace-go/blob/main/ddtrace/ext/span_kind.go.

Thus, code here should like follows
``` go
	switch span.Meta["span.kind"] {
	case "server":
		newSpan.SetKind(ptrace.SpanKindServer)
	case "client":
		newSpan.SetKind(ptrace.SpanKindClient)
	case "producer":
		newSpan.SetKind(ptrace.SpanKindProducer)
	case "consumer":
		newSpan.SetKind(ptrace.SpanKindConsumer)
	case "internal":
		newSpan.SetKind(ptrace.SpanKindInternal)
	default:
		newSpan.SetKind(ptrace.SpanKindUnspecified)
	}
```

_Originally posted by @moonheart in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/5836#discussion_r1091397100_

**Comment by boostchicken** [comment]
This was addressed in the original PR #5836

**Comment by moonheart** [comment]
Hello @boostchicken , I think the following code `span.SpanType` is a misuse.
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/8dd67c11ec1a425516a61df75a9dacc248092d9a/receiver/datadogreceiver/translator.go#L74-L89

 `span.SpanType` is used internally in datadog with values from https://github.com/DataDog/dd-trace-go/blob/main/ddtrace/ext/app_types.go like `web`,`http`,`sql` etc.

According to https://github.com/DataDog/dd-trace-go/blob/main/contrib/README.md which was added later, 
> The span.kind tag should be set in root spans with either a client, server, producer, or consumer value according to the [definitions](https://github.com/DataDog/dd-trace-go/blob/main/ddtrace/ext/span_kind.go) found in the repository. If the value is determined to be internal, then omit the tag as that is the assumed default value. Otherwise, explicitly set it with a value from above.

We should check  `span.Meta["span.kind"]` first and if there is not `span.kind` meta, we convert `span.Type` to SpanKind.
